mern/client/src/Controller/recoverSessionController.js

```diff
@@ -0,0 +1,10 @@
+import { sendToDAO, getQAFromFlow } from "../Interactor/recoverSessionInteractor"
+
+// returns starting QA pair from session ID
+export async function recoverFlow(sessionID) {
+    // returns flow promise object
+    const flow = await sendToDAO(sessionID);
```

Comment:
This part seems a little weird. You are getting flow as a return, which is a promise here. Then you are sending flow into the getQAFromFlow function. This means you retrieved a response from the DAO, then sent it back to the DAO. 

I get that the goal is to get the first starting question, but realistically these shows be separated into different controllers and interactors. 

The way that you want to do it is, to pass the flow response all the way back up to where you need to use it. 
Then, access the first element of the questions list at the top level which gives you the id of the first question.
After, you go through a QA controller which then calls the QA interactor which then calls the QA DAO that fetches the QA object and returns it to the top layer.

Reply:
Oops, looks like the comments were out of date here. Thanks for the flag. The flow return is a json object, not a promise.

I see what you are saying about retrieving the QA object through a seperate controller/interactor/dao. I mistakenly thought the current_question parameter in the flow was returning the QA object, not just the ID. Will make the fix to retrieve the QA object itself!